### PR TITLE
Change omitted names of type-paratemers to understandable names and add some functions

### PR DESCRIPTION
**Pull request by junjihashimoto**

### CODE REVIEW

File: hasktorch/src/Torch/Static/Native.hs [hunk 1]
```diff
@@ -971,141 +1074,216 @@ logsumexp t = unsafePerformIO $ cast3 ATen.logsumexp_tlb t (natValI @dim) (keepO
...
+-- matrix_power _input _n = unsafePerformIO $ (cast2 ATen.matrix_power_tl) _input _n
+
 
--- max_values :: Tensor dtype shape -> Int -> Bool -> Tensor dtype shape
--- max_values _self _dim _keepdim = unsafePerformIO $ (cast3 ATen.max_values_tlb) _self _dim _keepdim
+type family DropValueWithKeepDim (shape :: [Nat]) (i :: Nat) (keepDim :: Bool) :: [Nat] where
+    DropValueWithKeepDim '[] _ _ = TypeError (Text "Can not find a element in the list.")
+    DropValueWithKeepDim (x: xs) 0 True = 1:xs
+    DropValueWithKeepDim (x: xs) 0 False = xs
+    DropValueWithKeepDim (x: xs) i b = x ': DropValueWithKeepDim xs (i-1) b
```
Comment (junjihashimoto) [inline hasktorch/src/Torch/Static/Native.hs]: I should remove this function. It is not used.

**Comment by tscholak** [comment]
hi @junjihashimoto do you think you can merge this soon?

**Comment by tscholak** [comment]
this is awesome btw!

**Comment by junjihashimoto** [comment]
@tscholak 
I think this pr can be merged soon.
But this pr has some conflict with https://github.com/hasktorch/hasktorch/pull/191.
Is it ok to merge this pr?

**Comment by junjihashimoto** [comment]
CI needs more memory.
https://circleci.com/gh/junjihashimoto/hasktorch/145
CircleCI's memory is 4gb. github-actions7s one is 8gb. So I think we should use github-actions.

**Comment by tscholak** [comment]
@junjihashimoto hm, I think we could merge #191 now if you like. There are some tests failing, but that can be sorted out quickly.

**Comment by tscholak** [comment]
The work on the transformer language model example is still very much a WIP, but so is the cnn example ;)

**Comment by tscholak** [comment]
I also support the move to github actions, they seem to be pretty good

**Comment by tscholak** [comment]
Btw, in my branch a changed the signature of dropout such that the input tensor is last. This makes it easier to chain it in expressions like `transpose @0 @1 . dropout 0.5 True . softmax @1`

**Comment by tscholak** [comment]
You may want to do that for the ops you worked on, too

**Comment by junjihashimoto** [comment]
Rebase and merge are done.

**Pull request merged**